gs_chunked_io/reader.py

```diff
@@ -44,7 +44,7 @@ def read(self, size: int=-1) -> bytes:
 
         ret_data = self._buffer[:size]
         del self._buffer[:size]
-        return ret_data
+        return bytes(ret_data)
```

Comment:
Casting to bytes unfortunately involves a copy of the underlying buffer, negatively affecting performance. See, [this Stack Overflow post for discussion](https://stackoverflow.com/questions/35880065/does-converting-from-bytearray-to-bytes-incur-a-copy).

However, we can still cast to bytes without increasing the number of buffer copies by taking a slice of a memoryview on line 45:
```
ret_data = bytes(memoryview(self._buffer)[:size])
```

Reply:
Sure, done!